scraper.rb

```diff
@@ -1,290 +1,46 @@
-  end
-
-  if page.has_xpath?('.//div[@class="dip_rojo"][contains(.,"Causó baja")]')
-    causo_baja = find(:xpath, './/div[@class="dip_rojo"][contains(.,"Causó baja")]')
-    end_date = causo_baja.text.match(/(\d+)\/(\d+)\/(\d+)\./).captures.reverse.join("-")
-  end
-
-  dob = ''
-  email = ''
-  twitter = ''
-  facebook = ''
-  photo = ''
-  within('div.titular_historico') do
-    dob = date_of_birth(all(:xpath, 'following::div/ul/li')[0].text)
-  end
-
-  # capybara doesn't support enough xpath to do this
-  # sensibly so we have to do this the longwinded way
-  if page.has_xpath?('//div[@class="webperso_dip"]/div[@class="webperso_dip_parte"|@class="webperso_dip_imagen"]/a')
-    all(:xpath, '//div[@class="webperso_dip"]/div[@class="webperso_dip_parte"|@class="webperso_dip_imagen"]/a').each do |link|
-      href = link['href']
-      if href.match(/mailto/)
-        email = link.text.tidy
-      end
-      if href.match(/twitter.com/)
-        twitter = href.match(/twitter.com\/(.*)$/).captures[0]
-      end
-      if href.match(/facebook.com/)
-        facebook = href
-      end
-    end
-  end
-
-  all('div#datos_diputado').each do |img|
-    within(img) do
-      if img.has_xpath?('.//p[@class="logo_group"]/img[@name="foto"]')
-        photo = find(:xpath, './/p[@class="logo_group"]/img[@name="foto"]')['src'].text
-      end
-    end
-  end
-
-  data = {
-    iddiputado: iddiputado,
-    name: "#{given_names} #{family_names}",
-    sort_name: name,
-    given_name: given_names,
-    family_name: family_names,
-    gender: gender_from(seat),
-    party: find('div#datos_diputado p.nombre_grupo').text.tidy,
-    faction_id: faction_id,
-    faction: faction,
-    source: url.to_s,
-    dob: dob,
-    term: term,
-    start_date: start_date,
-    end_date: end_date,
-    email: email,
-    twitter: twitter,
-    facebook: facebook,
-    phone: all('div.texto_dip').map(&:text).join('').match(/Teléfono: (.*)$/).to_a.last.to_s.tidy,
-    fax: all('div.texto_dip').map(&:text).join('').match(/Fax: (.*)$/).to_a.last.to_s.tidy,
-    constituency: seat[/Diputad. por (.*)\./, 1],
-    photo: photo,
-  }
-  data[:photo] = URI.join(url, data[:photo]).to_s unless data[:photo].to_s.empty?
-
-  all_terms_url = find('div.soporte_year li a')['href'].match('.*listadoFichas.*').to_a.first.to_s
-
-  # it might seem a bit odd to do this only once we've worked out everything
-  # else but doing it this way means we don't need to visit the all terms page
-  # and then go back so it's one less network call per person
-  id = get_unique_id(all_terms_url, term, iddiputado, name)
-
-  # don't save things if we don't get an id
-  if id.nil?
-    #puts "no id so not saving"
-      return
-  end
-
-  data[:id] = id
-
-  #puts "%s - %s\n" % [ data[:name], data[:id] ]
-  ScraperWiki.save_sqlite([:id, :term], data)
+ScraperWiki.sqliteexecute('DELETE FROM data') rescue nil
+merged_memberships.each do |mem|
+  ScraperWiki.save_sqlite([:name, :term], mem)
```

Comment:
Keying on `name` here seems wrong — can you guarantee that there have never been two members with the same name? 

(And can a Member only have a single membership per term?)